cmd/config.go

```diff
@@ -0,0 +1,121 @@
+
+Aliases:
+  {{.NameAndAliases}}
+{{end}}{{if .HasExample}}
+
+Examples:
+{{ .Example }}{{end}}{{ if .HasAvailableSubCommands}}
+
+Available Commands:{{range .Commands}}{{if .IsAvailableCommand}}
+{{rpad .Name .NamePadding }} {{.Short}}{{end}}{{end}}{{end}}{{ if .HasAvailableLocalFlags}}
+
+Available Parameters: UpdateNotification
+
+Flags:
+{{.LocalFlags.FlagUsages | trimRightSpace}}{{end}}{{ if .HasAvailableInheritedFlags}}
+
+Global Flags:
+{{.InheritedFlags.FlagUsages | trimRightSpace}}{{end}}{{if .HasHelpSubCommands}}
+
+Additional help topics:{{range .Commands}}{{if .IsHelpCommand}}
+  {{rpad .CommandPath .CommandPathPadding}} {{.Short}}{{end}}{{end}}{{end}}{{ if .HasAvailableSubCommands }}
+
+Use "{{.CommandPath}} [command] --help" for more information about a command.{{end}}
+`
+
+// configurationCmd represents the app command
+var configurationCmd = &cobra.Command{
+	Use:   "config",
+	Short: "Modifies configuration settings",
+	Long:  `Modifies Odo specific configuration settings within the config file.`,
+	Example: fmt.Sprintf("%s\n%s\n",
+		configurationGetCmd.Example,
+		configurationSetCmd.Example),
+	Aliases: []string{"configuration"},
+	// 'odo config' is the same as 'odo config get'
+	Args: func(cmd *cobra.Command, args []string) error {
+		if len(args) >= 1 && args[0] != "get" && args[0] != "set" {
+			return fmt.Errorf("Unknown command, use set or get")
+		}
+		return nil
+	}, Run: func(cmd *cobra.Command, args []string) {
+		if len(args) > 0 && args[0] == "set" {
+			configurationSetCmd.Run(cmd, args)
+		} else if len(args) > 0 && args[0] == "get" {
+			configurationGetCmd.Run(cmd, args)
+		} else {
+			configurationGetCmd.Run(cmd, args)
+		}
+	},
+}
+
+var configurationSetCmd = &cobra.Command{
+	Use:   "set",
+	Short: "Set a value in odo config file",
+	Long:  "Set an individual value in the Odo configuration file",
+	Example: `  # Set a configuration value
+   odo config set UpdateNotification false
+ 	`,
+	Args: func(cmd *cobra.Command, args []string) error {
+		if len(args) < 2 {
+			return fmt.Errorf("Please provide a parameter name and value")
+		} else if len(args) > 2 {
+			return fmt.Errorf("Only one value per parameter is allowed")
+		} else {
+			return nil
+		}
+	}, Run: func(cmd *cobra.Command, args []string) {
+		cfg, err := config.New()
+		if err != nil {
+			fmt.Println(err, ": unable to set configuration")
+		}
+		value, err := strconv.ParseBool(args[1])
+		if err != nil {
+			fmt.Println(err, ": unable to set configuration")
+		}
+		cfg.SetConfiguration(strings.ToLower(args[0]), value)
+	},
+}
+
+var configurationGetCmd = &cobra.Command{
+	Use:   "get",
+	Short: "View current value",
+	Long:  "View current configuration",
+	Example: `  # Get the current configuration
+   odo config get
+ 	`,
+	Args: cobra.ExactArgs(0),
+	Run: func(cmd *cobra.Command, args []string) {
+		cfg, err := config.New()
+		if err != nil {
+			fmt.Println(err, ": unable to get configuration")
+		}
+		cfg.GetConfiguration()
```

Comment:
configuration should be printed in nice table, something similar to what we are doing with all `list` commands